lib/plugins/features/movies/controller.js

```diff
@@ -8,6 +8,32 @@ exports.create = async (payload) => {
   Reflect.deleteProperty(payload, 'title');
 
   const movie = await new Movie().save(payload);
-
   return new Movie({ id: movie.id }).fetch();
 };
+
+exports.list = async (query) => {
+
+  return new Movie().query((qb) => {
+
+    // filter exact year
+    if (query.year) {
+      qb.where('release_year', '=', query.year);
+    }
+
+    // filter between 2 years
+    if (query.from_year) {
+      qb.where('release_year', '>=', query.from_year);
+    }
+
+    if (query.to_year) {
+      qb.where('release_year', '<=', query.to_year);
+    }
+
+    // fuzzy filter by title
+    if (query.title) {
+      qb.where('name', 'ILIKE', `%${query.title}%`);
+    }
+
+  }).fetchAll();
+
+};
```

Comment:
nit: I'd recommend removing the new lines around this `return` and also at the beginning and end of its query function. In general, you want to be careful about using _too_ many newlines! They're useful when separating the `if` blocks, but unnecessary when surrounding the contents of a function.